### PR TITLE
Use redux in components

### PR DESCRIPTION
**Pull request by iLynette**
In this project, i implemented the following;

- [ ] kept state immutable in reducers!

- [ ] Import Redux Provider and your store in the main component of your React app.

- [ ] Connect your app to the Redux store.

- [ ] Make sure that list of books displays books from Redux store.

- [ ] Use useSelector to consume the state.

- [ ] In your React component responsible for adding new books, add funcionality for "Add new" button click:

- [ ] Import Redux useDispatch.

- [ ] Set the data inputs in the local React state (set title and author - NOTE: categories and comments are NOT part of this step).

- [ ] Once your new book object is ready to be submitted to Redux store, dispatch a corresponding action (import Action Creator that you created before and use it here).

- [ ] In your React component responsible for removing books, add a funcionality for "Remove" button click:

- [ ] Import Redux useDispatch.

- [ ] Dispatch a corresponding action (import Action Creator that you created before and use it here).

- [ ] In your React component responsible for categories, add a funcionality for "Check status" button click:

- [ ] Import Redux useDispatch.

- [ ] Dispatch a corresponding action (import Action Creator that you created before and use it here).

### CODE REVIEW

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @iLynette, :wave:

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️

- [ ] Following the project requirements, The component `Categories` should contain a button `Check status` for checking the status of the books.

* this button should use a function made in the component `Categories` that dispatches the `underConstruction` action creator with the id as its argument.
Check out the requirements below 👇:

image

_See the inline comments below for more required changes_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Categories.js [hunk 1]
```diff
@@ -1,7 +1,16 @@
 import React from 'react';
+import { useDispatch } from 'react-redux';
+import { status } from '../redux/categories/categories';
 
-const Categories = () => (
-  <div>Nothing to see here</div>
-);
+const Categories = () => {
```
Comment (Meltrust) [inline src/components/Categories.js:5]: - [ ] OPTIONAL:  Good job adding the "check status" button and dispatching its corresponding action! I would suggest you actually render the status when pressing that button like this:

```JAVASCRIPT
const Categories = () => {
  const dispatch = useDispatch();
  const stats = useSelector((state) => state.categoriesReducer);
  return (
    <div>
      <button type="button" className="submitBtn" onClick={() => dispatch(status())}>
        Check Status
      </button>
      {stats}
    </div>
  );
};
```

That way, the status will be rendered! 💪

**Review by Meltrust — APPROVED** [review]
Hi @iLynette,


## Wow, you did it 🎉
Brilliant
## Thank you for implementing the 'check status' button' that dispatches a Redux action, as requested by the previous reviewer 💪 🥇 ㊗️ 
### Well done!




Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Categories.js (hunk 1 above)
Comment (iLynette) [reply src/components/Categories.js:5]: Thank you very much @Meltrust I will implement this

**Pull request merged**